pkgs/dart_mcp/test/resources_support_test.dart

```diff
@@ -136,6 +136,81 @@ void main() {
     await environment.shutdown();
   });
 
+  test('resource change notifications are throttled', () async {
+    final environment = TestEnvironment(
+      TestMCPClient(),
+      (c) => TestMCPServerWithResources(channel: c),
+    );
+    await environment.initializeServer();
+
+    final serverConnection = environment.serverConnection;
+    final server = environment.server;
+
+    final resourceListChangedQueue = StreamQueue(
+      serverConnection.resourceListChanged,
+    );
+
+    final resources = [
+      for (var i = 0; i < 5; i++) Resource(name: '$i', uri: 'foo://$i'),
+    ];
```

Comment:
nit: using List.generate might be more terse

Reply:
see my comment below, not sure which I like more

Reply:
Readability guidelines generally prefer using collection literals, I think this looks good as is.